CARDS-2656: Refactor login related components to functional

veronikaslc opened the pull request:
Recommended codereview mode with `Hide whitespace` setting on

veronikaslc committed:
CARDS-2656: Refactor login related components to functional

marta- commented on modules/login/src/main/frontend/src/login/loginForm.js:
```diff
@@ -146,118 +127,129 @@ class SignIn extends React.Component {
               if (loginPopup.closed === true) {
                 clearInterval(checkLoginTimer);
-                this.props.handleLogin && this.props.handleLogin(true);
+                handleLogin && handleLogin(true);
```
or 
```suggestion
                handleLogin?.(true);
```

veronikaslc committed:
CARDS-2656: Refactor login related components to functional
Better codestyle and error handling

veronikaslc commented:
@marta- Do we want to also refactor loginDialogue -> loginDialog?

veronikaslc committed:
CARDS-2656: Refactor login related components to functional
Refactored loginDialogue to loginDialog

marta- commented:
Is `ChangeUserPasswordDialogue` related enough to login to be refactored and renamed in this PR or will there be a separate PR for it? I vote for including it.

veronikaslc committed:
CARDS-2656: Refactor login related components to functional
Apply prop types check wrap

veronikaslc commented:
In testing

marta- commented:
@veronikaslc there are references to the `loginDialogue` asset in this recently merged PR to https://github.com/data-team-uhn/cards/pull/1974/files that would likely need to be renamed in this PR.

veronikaslc commented:
@marta- thats why I commented "in testing" to indicate this PR is not ready for review yet

veronikaslc commented:
Rebased, retested, ready for review

marta- commented on modules/login/src/main/frontend/src/login/loginForm.js:
```diff
@@ -33,55 +35,39 @@ import VisibilityIcon from '@mui/icons-material/Visibility';
-      }
-    });
-  }
+      .then((resp) => setSingleStepEntry(!!resp.ok));
```
This does the opposite than before.
```suggestion
      .then((resp) => setSingleStepEntry(!resp.ok));
```

marta- commented on modules/login/src/main/frontend/src/login/signUpForm.js:
```diff
@@ -214,48 +215,42 @@ class SignUpForm extends React.Component {
 
-        this.props.handleSuccess && this.props.handleSuccess();
-        this.props.loginOnSuccess && this.signIn(username, password);
+        handleSuccess && handleSuccess();
```
Since we're making things shorter...
```suggestion
        handleSuccess?.();
```

veronikaslc committed:
CARDS-2656: Refactor login related components to functional

Correcred response status logic
Code optimisation
Better login error wording
Better codestyle
Shorter syntax
Better name for the re-login dialog

marta- commented on modules/login/src/main/frontend/src/login/loginDialog.js:
```diff
@@ -42,20 +43,22 @@ export function fetchWithReLogin(displayLoginCtx, url, fetchArgs, discardOnFailu
-  constructor(props) {
-    super(props);
-  }
+function DialogLoginContainer(props) {
```
What an awkward component name! Wdyt about renaming it to LoginPopup (if I understand correctly what it does)? @veronikaslc @sdumitriu

veronikaslc replied:
Rename to `RelogingDialog`

marta- replied:
`ReLoginDialog`

marta- commented on modules/login/src/main/frontend/src/login/loginForm.js:
```diff
@@ -145,114 +128,128 @@ class SignIn extends React.Component {
+                        <IconButton
+                          size="large"
+                          aria-label="Toggle password visibility"
+                          onClick={togglePasswordMask}
```
I would get rid of the `togglePasswordMask` function since it's only used here.

```suggestion
                          onClick={() => setPasswordIsMasked(!passwordIsMasked)}
```

marta- commented on modules/login/src/main/frontend/src/login/loginForm.js:
```diff
@@ -145,114 +128,128 @@ class SignIn extends React.Component {
             let checkLoginTimer = setInterval(() => {
               if (loginPopup.closed === true) {
                 clearInterval(checkLoginTimer);
-                this.props.handleLogin && this.props.handleLogin(true);
+                handleLogin?.(true);
               }
             }, 1000);
```
I'm pretty sure the entire `else` branch (lines 117-133) is dead code. and got replaced by the `DialogLoginContainer` component.

marta- commented on modules/login/src/main/frontend/src/login/loginForm.js:
```diff
@@ -145,114 +128,128 @@ class SignIn extends React.Component {
           }
         })
-        .catch((err) => this.setState({failedLogin: "Error occurred while handling third-party identity provider"}))
+        .catch((err) => setFailedLogin("Error occurred while handling third-party identity provider"));
```
While we're here, let's make this more english: 
```suggestion
        .catch((err) => setFailedLogin("An error occurred while handling the third-party identity provider."));
```

marta- commented on modules/login/src/main/frontend/src/login/loginMainComponent.js:
```diff
@@ -26,63 +26,56 @@ import Logo from "../components/Logo";
+              variant="outlined"
+              fullWidth
+              className={classes.main}
+              onClick={handleSwap}
```
I would get rid of `handleSwap` as it's only used in this one place.
```suggestion
              onClick={() => setSignInShow(!signInShown)}
```

marta- commented on modules/login/src/main/frontend/src/login/loginMainComponent.js:
```diff
@@ -26,63 +26,56 @@ import Logo from "../components/Logo";
+          <SignIn handleLogin={handleLogin} redirectOnLogin={redirectOnLogin}/>
+          :
+          <SignUpForm loginOnSuccess={true} handleLogin={handleLogin} />
+        }
```
So much easier to read!

veronikaslc commented:
Resolved conflicts and rebased on the latest dev, squashed last 6 code optmisation commits for cleaner github history.

marta- commented on modules/login/src/main/frontend/src/login/loginDialog.js:
Rename this file to ReLoginDialog.jsx to match the name of the default exported component.

marta- commented on modules/login/src/main/frontend/src/login/loginForm.js:
Rename this file to SignInForm.jsx,  and the default exported component to `SignInForm`.

marta- replied:
According team team discussion - rename SignInForm to LoginForm

marta- commented on modules/login/src/main/frontend/src/login/loginMainComponent.js:
Rename this file to MainLoginContainer.jsx to match the name of the default exported component.

marta- commented on modules/login/src/main/frontend/src/login/signUpForm.js:
Rename this file to SignUpForm.jsx to match the name of the default exported component.

marta- replied:
According team team discussion - rename SignUpForm to RegistrationForm

marta- reviewed (commented):
Files that export default components should have the same name as the exported component, (including capitalization).

Question before the renaming is implemented: We use both `login` and `signin` terms. Specifically, the `loginForm.jsx` file exports the `SignIn` component. We also have the `signUpForm` file exporting the `SignUpForm` component. Do we want to keep "singin"/"signup" for these two components and "login" everywhere else, or do we want to turn "signin" into "login"? @data-team-uhn/lfs-devs

veronikaslc committed:
CARDS-2656: Refactor login related components to functional
Renamed login components

veronikaslc committed:
CARDS-2656: Refactor login related components to functional
Added back missed signUpEnabled (unused for now);
Better alignment and codestyle
